others/iterative_tree_traversals.cpp

```diff
@@ -0,0 +1,362 @@
+
+    // Calling postOrderIterative() function by passing a root node,
+    // and storing the postorder traversal in result.
+    result = binaryTree.postOrderIterative(root);
+
+    // Self-testing the result using `assert`
+    for(int i = 0; i < result.size(); i++)
+        assert(actual_result[i] == result[i]);
+
+    // Printing the result storing postorder.
+    std::cout<< "\nPostOrder Traversal Is : "<< std::endl;
+    for(auto i: result) {
+        std::cout<< i<< "  ";
+    }
+}
+
+/**
+ * @brief Test the computed inorder with the actual inorder on negative value.
+ * @param binaryTree instance of BinaryTree class
+ * @param root head/root node of a tree
+ */
+static void test6(others::iterative_tree_traversals::BinaryTree binaryTree, others::iterative_tree_traversals::Node *root){
+    std::vector<int64_t> actual_result{-4, -2, -5, -1, -3};
+    std::vector<int64_t> result; ///< result stores the inorder traversal of the binary tree.
+
+    // Calling inOrderIterative() function by passing a root node,
+    // and storing the inorder traversal in result.
+    result = binaryTree.inOrderIterative(root);
+
+    // Self-testing the result using `assert`
+    for(int i = 0; i < result.size(); i++)
+        assert(actual_result[i] == result[i]);
+
+    // Printing the result storing inorder.
+    std::cout<< "\nInOrder Traversal Is : "<< std::endl;
+    for(auto i: result) {
+        std::cout<< i<< "  ";
+    }
+}
+
+/**
+ * @brief Main function
+ * @returns 0 on exit
+ */
+int main() {
+    // Creating a tree with the following structure,
+    /* 
+              1
+            /   \
+           2     3
+         /   \
+        4     5
+    */
+
+    others::iterative_tree_traversals::BinaryTree binaryTree; ///< instace of BinaryTree, used to access its members functions.
+    others::iterative_tree_traversals::Node *root = binaryTree.createNewNode(1);
+    root->left = binaryTree.createNewNode(2);
+    root->right = binaryTree.createNewNode(3);
+    root->left->left = binaryTree.createNewNode(4);
+    root->left->right = binaryTree.createNewNode(5);
+
+    std::cout<< "\n| Tests for positive data value |"<< std::endl;
+    test1(binaryTree, root);  // run preorder-iterative test
+    std::cout<< "\nPre-order test Passed!"<< std::endl;
+
+    test2(binaryTree, root);  // run postorder-iterative test
+    std::cout<< "\nPost-order test Passed!"<< std::endl;
+
+    test3(binaryTree, root);  // run inorder-iterative test
+    std::cout<< "\nIn-order test Passed!"<< std::endl;
+
+    // Modifying tree for negative values.
+    root->data = -1;
+    root->left->data = -2;
+    root->right->data = -3;
+    root->left->left->data = -4;
+    root->left->right->data = -5;
+
+    std::cout<< "\n| Tests for negative data values |"<< std::endl;
+    test4(binaryTree, root);  // run preorder-iterative test on negative values
+    std::cout<< "\nPre-order test on negative value Passed!"<< std::endl;

```

Comment:
IMO, this makes it look cooler. 🙂
```suggestion
    std::cout<< "\nPre-order test on-negative value Passed!"<< std::endl;
```